app/controllers/api/v1/migrations/external_controller.rb

```diff
@@ -90,21 +110,78 @@ def create_room
+          end
+
+          # Creates the associated SharedAccess
+          room_hash[:shared_users_emails].any? && room_hash[:shared_users_emails].each do |email|
+            user = User.find_by(email:)
+            return render_error status: :bad_request unless user
+
+            SharedAccess.create!(room_id: room.id, user_id: user.id)
+          end
+
+          render_data status: :created
+        end
+
+        # POST /api/v1/migrations/site_settings.json
+        # Expects: { settings: { site_settings: { :PrimaryColor, :PrimaryColorLight, :PrimaryColorDark,
+        #                          :Terms, :PrivacyPolicy, :RegistrationMethod, :ShareRooms, :PreuploadPresentation },
+        #                        room_configurations: { :record, :muteOnStart, :guestPolicy, :glAnyoneCanStart,
+        #                          :glAnyoneJoinAsModerator, :glRequireAuthentication } } }
+        # Returns: { data: Array[serializable objects] , errors: Array[String] }
+        # Does: Creates a SiteSettings or a RoomsConfiguration.
+        def create_site_settings
+          settings_hash = site_settings_params.to_h
+
+          render_data status: :created unless settings.any?
+
+          settings_hash[:site_settings].any? && settings_hash[:site_settings].each do |name, value|
+            site_setting = SiteSetting.joins(:setting).find_by('settings.name': name, provider: 'greenlight')
+            return render_error status: :bad_request unless site_setting
+
+            site_setting.update!(value:)
+            return render_error status: :bad_request unless site_setting.save
+          end
+
+          settings_hash[:room_configurations].any? && settings_hash[:room_configurations].each do |name, value|
+            room_configuration = RoomsConfiguration.joins(:meeting_option).find_by('meeting_options.name': name, provider: 'greenlight')
+            return render_error status: :bad_request unless room_configuration
+
+            room_configuration.update!(value:)
+            return render_error status: :bad_request unless room_configuration.save
+          end
+
           render_data status: :created
         end
 
         private
 
         def role_params
-          decrypted_params.require(:role).permit(:name)
+          decrypted_params.require(:role).permit(:name, role_permissions: {})
         end
 
         def user_params
           decrypted_params.require(:user).permit(:name, :email, :external_id, :language, :role)
         end
 
         def room_params
-          decrypted_params.require(:room).permit(:name, :friendly_id, :meeting_id, :last_session, :owner_email)
+          decrypted_params.require(:room).permit(:name, :friendly_id, :meeting_id, :last_session, :owner_email, room_settings: {},
+                                                                                                                shared_users_emails: [])
+        end
```

Comment:
Having an empty hash or and empty array in permit is not optimal but I believe that specifying every nested element would require a refactoring of the data structure.